src/game/client/tf/vgui/tf_clientscoreboard.cpp

```diff
@@ -513,10 +513,12 @@ void CTFClientScoreBoardDialog::OnCommand( const char *command )
 				if ( !pIssueKeyValues )
 					return;
 
-				int playerIndex = pIssueKeyValues->GetInt( "playerIndex", 0 );
-				if ( playerIndex > 0 && playerIndex <= MAX_PLAYERS )
+				const char* playerName = pIssueKeyValues->GetString("name", NULL);
+				if (playerName)
 				{
-					engine->ClientCmd_Unrestricted( VarArgs( "spec_player %d\n", playerIndex ) );
+					char FixedPlayerName[128];
+					V_snprintf(FixedPlayerName, strlen(playerName) + 2 + 1, "\"%s\"", playerName);
```

Comment:
This size calculation is incorrect, should be `sizeof(FixedPlayerName)`
You are formatting the string twice which is unnecessary. Just format it as `"spec_player \"%s\""` directly
Also rename `FixedPlayerName` to `szFixedPlayerName` to follow Valve conventions, and add spacing around your brackets